crates/trie/src/trie_cursor/subnode.rs

```diff
@@ -75,48 +71,34 @@ impl CursorSubNode {
 
     /// Returns `true` if the state flag is set for the current nibble.
     pub fn state_flag(&self) -> bool {
-        if let Some(node) = &self.node {
-            if self.nibble >= 0 {
-                return node.state_mask.is_bit_set(self.nibble as u8)
-            }
-        }
-        true
+        self.node
+            .as_ref()
+            .map_or(true, |node| self.nibble >= 0 && node.state_mask.is_bit_set(self.nibble as u8))
     }
 
     /// Returns `true` if the tree flag is set for the current nibble.
     pub fn tree_flag(&self) -> bool {
-        if let Some(node) = &self.node {
-            if self.nibble >= 0 {
-                return node.tree_mask.is_bit_set(self.nibble as u8)
-            }
-        }
-        true
+        self.node
+            .as_ref()
+            .map_or(true, |node| self.nibble < 0 || node.tree_mask.is_bit_set(self.nibble as u8))
```

Comment:
let's avoid relying on short-circuit evaluation since it makes the checks less obvious 
```suggestion
        self.node
            .as_ref()
            .map_or(true, |node| self.nibble >= 0 && node.tree_mask.is_bit_set(self.nibble as u8))
```